spec/hoteladmin_spec.rb

```diff
@@ -0,0 +1,272 @@
+require_relative 'spec_helper'
+
+describe "HotelAdmin" do
+  let (:hotel) {
+     HotelAdmin.new
+  }
+  describe "HotelAdmin#Initialize" do
+    it "can create a hoteladmin object" do
+      expect(hotel).must_be_instance_of HotelAdmin
+    end
+
+    it "initializes with readable rooms and reservations array attributes" do
+      expect(hotel.rooms).must_be_instance_of Array
+      expect(hotel.rooms.length).must_equal 20
+      expect(hotel.rooms.first).must_be_instance_of Room
+      expect(hotel.reservations).must_be_instance_of Array
+      expect(hotel.reservations).must_be_empty
+    end
+  end
+
+  describe "HotelAdmin#build_rooms" do
+    it "creates our room objects numbered 1-20" do
+      expect(hotel.build_rooms.length).must_equal 20
+      expect(hotel.build_rooms.first).must_be_instance_of Room
+      expect(hotel.build_rooms).must_be_instance_of Array
+    end
+  end
+
+  describe "HotelAdmin#list_rooms" do
+    it "will return the array room objects" do
+          # 20.times do |index|
+          #   hotel.rooms << Room.new(index+1)
+          # end
+          expect(hotel.list_rooms[0]).must_be_instance_of Room
+          expect(hotel.list_rooms).must_be_instance_of Array
+          expect(hotel.list_rooms.length).must_equal 20
+    end
+  end
+
+  describe "HotelAdmin#retrieve_by_date" do
+    before do
+      hotel.reservations << Reservation.new({guest_id: "SoccerMom2010@gmail.com", room: 1, date_range: (Date.new(2018,10,20)...Date.new(2018,10,22))})
+      hotel.reservations << Reservation.new({guest_id: "Guccifer2.0@ada.com", room: 1, date_range: (Date.new(2018,12,02)...Date.new(2018,12,07))})
+      hotel.reservations << Reservation.new({guest_id: "Jfahmy07@gmail.com", room: 2, date_range: (Date.new(2018,12,03)...Date.new(2018,12,06))})
+    end
+
+    it "raises an argument error if invalid date object is provided" do
+      expect{ (hotel.retrieve_by_date("I am a string")) }.must_raise ArgumentError
+    end
+
+    it "returns an array of the reservations relevant to given date" do
```

Comment:
You should also test that it returns an empty array if there are no reservations on that date.

Reply:
Fixed.